hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/DataSerializableSerializer.java

```diff
@@ -62,14 +62,18 @@ final class DataSerializableSerializer implements StreamSerializer<DataSerializa
     DataSerializableSerializer(Map<Integer, ? extends DataSerializableFactory> dataSerializableFactories,
                                ClassLoader classLoader) {
         try {
-            final Iterator<DataSerializerHook> hooks = ServiceLoader.iterator(DataSerializerHook.class, FACTORY_ID, classLoader);
+            Iterator<DataSerializerHook> hooks = ServiceLoader.iterator(DataSerializerHook.class, FACTORY_ID, classLoader);
             while (hooks.hasNext()) {
                 DataSerializerHook hook = hooks.next();
                 final DataSerializableFactory factory = hook.createFactory();
                 if (factory != null) {
                     register(hook.getFactoryId(), factory);
                 }
             }
+            hooks = ServiceLoader.iterator(DataSerializerHook.class, FACTORY_ID, classLoader);
```

Comment:
This seems like a costly operation. Instead of loading all the hooks again, can we just store the hooks loaded in the above loop in a list and iterate and call this afterFactoriesCreated on that list?